File: Warm-up-Challenges/counting_valleys.rb
def counting_valleys(steps, path)
  valleys = 0
  level = 0
  
  path.split('').each do |step|
    if step == 'U' && level == -1
      valleys += 1
    elsif step == 'U'
      level += 1
    else
      level -= 1
    end
  end
  p valleys
end

counting_valleys(8, 'UDDDUDUU') # output 1
counting_valleys(12, 'DDUUDDUDUUUD') # output 2
counting_valleys(100, 'DDUUDDDUDUUDUDDDUUDDUDDDUDDDUDUUDDUUDDDUDDDUDDDUUUDUDDDUDUDUDUUDDUDUDUDUDUUUUDDUDDUUDUUDUUDUUUUUUUUU') # output 2

File: Warm-up-Challenges/counting_valleys.js
const countingValleys = (steps, path) => {
  let valleys = 0;
  let level = 0;
  const pathArr = path.split('')

  for (let i = 0; i < steps; i++) {
    if (pathArr[i] === 'U' && level === -1) {
      valleys += 1;
    }
  
    if (pathArr[i] === 'U') {
      level += 1
    } else {
      level -= 1
    }
  };
  console.log(valleys);
};

countingValleys(8, 'UDDDUDUU') // output 1
countingValleys(12, 'DDUUDDUDUUUD') // output 2
countingValleys(100, 'DDUUDDDUDUUDUDDDUUDDUDDDUDDDUDUUDDUUDDDUDDDUDDDUUUDUDDDUDUDUDUUDDUDUDUDUDUUUUDDUDDUUDUUDUUDUUUUUUUUU') // output 2
